internal/cmd/list-alerts.go

```diff
@@ -0,0 +1,262 @@
+			}
+
+			for _, incident := range incidents {
+				bunt.Printf("\n%d. *%s*\n", numPrintedIncidents+1, incident.Title)
+
+				if incident.Description != incident.Title {
+					bunt.Printf("   *Description:* \n")
+					bunt.Printf("   %s\n", incident.Description)
+				}
+
+				bunt.Printf("   *Link:* %s\n", incident.Self)
+
+				start, err := time.Parse("2006-01-02T15:04:05Z", incident.CreatedAt)
+				if err != nil {
+					return err
+				}
+
+				end, err := time.Parse("2006-01-02T15:04:05Z", incident.LastStatusChangeAt)
+				if err != nil {
+					return err
+				}
+
+				bunt.Printf("   *Time:* %s - %s (%s)\n",
+					start.Local().Format("2006-01-02 15:04:05"),
+					end.Local().Format("2006-01-02 15:04:05"),
+					end.Sub(start),
+				)
+
+				notes, err := client.ListIncidentNotes(incident.Id)
+				if err != nil {
+					return err
+				}
+				if len(notes) > 0 {
+					bunt.Printf("   *Notes:*\n")
+					for j := len(notes) - 1; j >= 0; j-- {
+						bunt.Printf("      %d. %s (by _%s_ at _%s_)\n",
+							len(notes)-j,
+							notes[j].Content,
+							lookUpNameByUserID(client, notes[j].User.ID),
+							formatNoteTime(notes[j].CreatedAt),
+						)
+					}
+				}
+
+				numPrintedIncidents++
+			}
+			if err != nil {
+				return err
+			}
+
+			if a.More {
+				offset += limit
+			} else {
+				moreIncidents = false
+			}
+		}
+
+		bunt.Println()
+
+		return nil
+	},
+}
+
+func lookUpNameByUserID(client *pagerduty.Client, id string) string {
+	user, err := client.GetUser(id, pagerduty.GetUserOptions{})
+	if err != nil {
+		return "unknown"
+	}
+	return user.Name
+}
+
+func formatNoteTime(input string) string {
+	time, err := time.Parse("2006-01-02T15:04:05-07:00", input)
+	if err != nil {
+		return input
+	}
+
+	return time.Local().Format("2006-01-02 15:04:05")
+}
+
+func filterIncidentsByTeamNameAndTime(incidents []pagerduty.Incident, oncalls []pagerduty.OnCall) ([]pagerduty.Incident, error) {
+	var filteredIncidents []pagerduty.Incident
+	for _, incident := range incidents {
+		for _, team := range incident.Teams {
+			for _, oncall := range oncalls {
+				isInTimeSpan, err := inTimeSpan(oncall.Start, oncall.End, incident.CreatedAt)
+				if err != nil {
+					return filteredIncidents, err
+				}
+				if isInTimeSpan && strings.Contains(oncall.Schedule.Summary, team.Summary) {
+					filteredIncidents = append(filteredIncidents, incident)
+					break
+				}
+			}
+		}
+	}
+	return filteredIncidents, nil
+}
+
+func inTimeSpan(s, e, check string) (bool, error) {
+	start, err := time.Parse("2006-01-02T15:04:05Z", s)
+	if err != nil {
+		return false, err
+	}
+	end, err := time.Parse("2006-01-02T15:04:05Z", e)
+	if err != nil {
+		return false, err
+	}
+	time, err := time.Parse("2006-01-02T15:04:05Z", check)
+	if err != nil {
+		return false, err
+	}
+	if start.Before(end) {
+		return !time.Before(start) && !time.After(end), nil
+	}
+	if start.Equal(end) {
+		return time.Equal(start), nil
+	}
+	return !start.After(time) || !end.Before(time), nil
```

Comment:
Comments would help to understand the ideas behind the three possible exits more quickly.